buku

```diff
@@ -6080,22 +6090,28 @@ POSITIONAL ARGUMENTS:
             bdb.close_quit(1)
 
         # Parse tags into a comma-separated string
-        tags = DELIM
-        keywords = args.add
-        if tags_in is not None:
-            if tags_in[0] == '+':
-                if len(tags_in) > 1:
-                    # The case: buku -a url tag1, tag2 --tag + tag3, tag4
-                    tags_in = tags_in[1:]
-                    # In case of add, args.add may have URL followed by tags
-                    # Add delimiter as url+tags may not end with one
-                    keywords = args.add + [DELIM] + tags_in
+        # --add may have URL followed by tags
+        keywords_except, keywords = [], args.add[1:]
+        # taglists are taken from --add (starting from 2nd value) and from --tags
+        # if taglist starts with '-', its contents are excluded from resulting tags
+        # if BOTH taglists is are either empty or start with '+'/'-', fetched tags are included
+        if keywords and keywords[0] == '-':
+            keywords, keywords_except = [], keywords[1:]
+        tags_add = (not keywords or keywords[0] == '+')
+        if tags_add:
+            keywords = keywords[1:]
+        if tags_in:
+            # note: need to add a delimiter as url+tags may not end with one
+            if tags_in[0] == '-':
+                keywords_except += [DELIM] + tags_in[1:]
+            elif tags_in[0] == '+':
+                keywords += [DELIM] + tags_in[1:]
             else:
-                keywords = args.add + [DELIM] + tags_in
-
-        if len(keywords) > 1:  # args.add is URL followed by optional tags
-            tags = parse_tags(keywords[1:])
+                keywords += [DELIM] + tags_in
+                tags_add = False
 
+        tags, tags_except = parse_tags(keywords), parse_tags(keywords_except)
+        tags, tags_except = ((s if s and s != DELIM else None) for s in [tags, tags_except])
```

Comment:
Replacing blank tag lists with `None` is optional, but I did it anyway for sake of consistency